File: LeetCode/1. Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> hash;
        vector<int> result;
        for(int i=0;i<nums.size();i++)
        {
            int need = target - nums[i];
            if(hash.find(need)!=hash.end())
            {
                result.push_back(hash[need]);
                result.push_back(i);
                return result;
            }
            hash[nums[i]]=i;
        }
        return result;
    }
};

File: LeetCode/1. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic = {}
        for i,value in enumerate(nums):
            if target-value in dic:
                return [i, dic[target-value]]
            else:
                dic[value]=i
        return []
